src/main/kotlin/bsu/cc/schedule/ConflictChecking.kt

```diff
@@ -0,0 +1,22 @@
+package bsu.cc.schedule
+
+import com.brein.time.timeintervals.collections.ListIntervalCollection
+import com.brein.time.timeintervals.indexes.IntervalTree
+import com.brein.time.timeintervals.indexes.IntervalTreeBuilder
+import com.brein.time.timeintervals.indexes.IntervalValueComparator
+
+fun buildScheduleIntervalTree(): IntervalTree {
+    val tree = IntervalTreeBuilder.newBuilder()
+            .collectIntervals { ListIntervalCollection() }
+            .build()
+
+    // required for overlap operations
+    tree.configuration.setValueComparator(IntervalValueComparator::compareInts)
+
+    // required for find operations
+    tree.configuration.setIntervalFilter { comp, val1, val2 ->
```

Comment:
So bizarre that this is required given that you already specified a value comparator.